src/main/scala/edu/berkeley/cs/rise/opaque/strategies.scala

```diff
@@ -105,6 +111,25 @@ object OpaqueOperators extends Strategy {
 
       filtered :: Nil
 
+    // Used to match non-equi joins
+    case Join(left, right, joinType, condition, hint) if isEncrypted(left) && isEncrypted(right) =>
+      // For perf reasons, `BroadcastNestedLoopJoinExec` prefers to broadcast left side if
+      // it's a right join, and broadcast right side if it's a left join.
+      // TODO: revisit it. If left side is much smaller than the right side, it may be better
+      // to broadcast the left side even if it's a left join.
+      val desiredBuildSide = if (joinType.isInstanceOf[InnerLike] || joinType == FullOuter)
+          getSmallerSide(left, right) else
+          getBroadcastSideBNLJ(joinType)
```

Comment:
Can you use the Spark SQL function `canBuildBroadcastLeft`?

Reply:
Not in Spark 3.0.1